demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1271 @@
+
+/**
+ * @brief Length of the payload stored in #payloadBuffer.
+ */
+static size_t payloadLength;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback to receive the incoming publish messages from the MQTT
+ * broker. Sets responseStatus if an expected CreateCertificateFromCsr or
+ * RegisterThing response is received, and copies the response into
+ * responseBuffer if the response is an accepted one.
+ *
+ * @param[in] pPublishInfo Pointer to publish info of the incoming publish.
+ * @param[in] packetIdentifier Packet identifier of the incoming publish.
+ */
+static void provisioningPublishCallback( MQTTPublishInfo_t * pPublishInfo,
+                                         uint16_t packetIdentifier );
+
+/**
+ * @brief Obtains the CSR from the file path in #PROVISIONING_CSR_PATH.
+ *
+ * @param[in] buffer Buffer into which to read the CSR.
+ * @param[in] bufferLength Length of #buffer.
+ * @param[out] outCsrLength The length of the CSR.
+ */
+static bool getCsr( char * buffer,
+                    size_t bufferLength,
+                    size_t * outCsrLength );
+
+/**
+ * @brief Creates the document to be published to the CreateCertificateFromCsr
+ * API in order to provision a certificate with the included CSR.
+ *
+ * @param[in] buffer Buffer into which to write the publish document.
+ * @param[in] bufferLength Length of #buffer.
+ * @param[out] outLengthWritten The length of the publish document.
+ */
+static bool generateCsrRequest( char * buffer,
+                                size_t bufferLength,
+                                size_t * outLengthWritten );
+
+/**
+ * @brief Creates the document to be published to the RegisterThing API in order
+ * to activate the provisioned certificate and receive a Thing name.
+ *
+ * @param[in] buffer Buffer into which to write the publish document.
+ * @param[in] bufferLength Length of #buffer.
+ * @param[in] certificateOwnershipToken The certificate's certificate ownership
+ * token.
+ * @param[in] certificateOwnershipTokenLength Length of #certificateOwnershipToken.
+ * @param[out] outLengthWritten The length of the publish document.
+ */
+static bool generateRegisterThingRequest( char * buffer,
+                                          size_t bufferLength,
+                                          const char * certificateOwnershipToken,
+                                          size_t certificateOwnershipTokenLength,
+                                          const char * serial,
+                                          size_t serialLength,
+                                          size_t * outLengthWritten );
+
+/**
+ * @brief Extracts the certificate, certificate ID, and certificate ownership
+ * token from a CreateCertificateFromCsr accepted response.
+ *
+ * @param[in] response The response document.
+ * @param[in] length Length of #response.
+ * @param[in] certificate The buffer to which to write the certificate.
```

Comment:
The description suggests that the data is copied into the buffer. If that's the case, it would be better to just update this to store pointer value of the starting address of the certificate in the response buffer. 
If the function is not copying the data, then this description needs to be updated

Reply:
The description suggests that the data is copied into the buffer. If that's the case, it would be better to just update this to store pointer value of the starting address of the certificate in the response buffer. 
If the function is not copying the data, then this description needs to be updated
```suggestion
 * @param[out] certificate This will be populated with the starting address of the certificate information
 * in @p response
```